packages/cursorless-engine/src/core/mergeSnippets.test.ts

```diff
@@ -0,0 +1,390 @@
+            {
+              body: ["user aaa"],
+              scope: {
+                langIds: ["typescript"],
+                scopeTypes: ["anonymousFunction"],
+              },
+            },
+          ],
+          description: "user snippet",
+        },
+      },
+    ],
+    expected: {
+      aaa: {
+        definitions: [
+          {
+            body: ["user aaa"],
+            scope: {
+              langIds: ["typescript"],
+              scopeTypes: ["anonymousFunction"],
+            },
+          },
+          {
+            body: ["someThirdParty aaa"],
+            scope: {
+              langIds: ["typescript"],
+              scopeTypes: ["anonymousFunction"],
+            },
+          },
+          {
+            body: ["core aaa"],
+            scope: {
+              langIds: ["typescript"],
+              scopeTypes: ["anonymousFunction"],
+            },
+          },
+        ],
+        description: "user snippet",
+      },
+    },
+  },
+
+  {
+    name: "should prefer more specific snippets, even if they are from a lower priority origin",
+    coreSnippets: {
+      aaa: {
+        definitions: [
+          {
+            body: ["core aaa"],
+            scope: {
+              langIds: ["typescript"],
+            },
+          },
+        ],
+        description: "core snippet",
+      },
+    },
+    userSnippets: [
+      {
+        aaa: {
+          definitions: [
+            {
+              body: ["user aaa"],
+            },
+          ],
+          description: "user snippet",
+        },
+      },
+    ],
+    expected: {
+      aaa: {
+        definitions: [
+          {
+            body: ["core aaa"],
+            scope: {
+              langIds: ["typescript"],
+            },
+          },
+          {
+            body: ["user aaa"],
+          },
+        ],
+        description: "user snippet",
+      },
+    },
+  },
+
+  {
+    name: "should prefer snippets with more specific languages",
+    coreSnippets: {
+      aaa: {
+        definitions: [
+          {
+            body: ["has no langIds"],
+          },
+          {
+            body: ["has two langIds"],
+            scope: {
+              langIds: ["typescript", "javascript"],
+            },
+          },
+          {
+            body: ["has one langId"],
+            scope: {
+              langIds: ["typescript"],
+            },
+          },
+        ],
+      },
+    },
+    expected: {
+      aaa: {
+        definitions: [
+          {
+            body: ["has one langId"],
+            scope: {
+              langIds: ["typescript"],
+            },
+          },
+          {
+            body: ["has two langIds"],
+            scope: {
+              langIds: ["typescript", "javascript"],
+            },
+          },
+          {
+            body: ["has no langIds"],
+          },
+        ],
+      },
+    },
+  },
+
+  {
+    name: "should prefer snippets with more specific languages, regardless of scopeType",
```

Comment:
this is not really true anymore.  although we sort them this way, we now still prefer the one with the scope when we actually get around to running the scope handler.  I should prob remove the code that looks at scope types at this point because it doesn't actually have any impact